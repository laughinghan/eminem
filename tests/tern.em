println (tern true ^(1) ^(2)) (tern null ^(3) ^(4))
println (tern (1.lt 2) ^(1) ^(2)) (tern (3.gt 4) ^(3) ^(4))
tern true ^(
  println "true :)"
) ^(
  println "false :("
)
tern null ^(
  println "true :)"
) ^(
  println "false :("
)
let .churchBooleanTrue (tern true)
let .churchBooleanFalse (tern null)
churchBooleanTrue ^(
  println "true :)"
) ^(
  println "false :("
)
churchBooleanFalse ^(
  println "true :)"
) ^(
  println "false :("
)
