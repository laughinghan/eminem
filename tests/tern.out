1
4
1
4
true :)
false :(
true :)
false :(
